tools/integration_tests/util/operations/file_operations.go

```diff
@@ -29,38 +29,59 @@ import (
 	"syscall"
 )
 
-func CopyFile(srcFileName string, newFileName string) (err error) {
-	if _, err = os.Stat(newFileName); err == nil {
-		err = fmt.Errorf("Copied file %s already present", newFileName)
-		return
+func copyFile(srcFileName, dstFileName string, allowOverwrite bool) (err error) {
+	if !allowOverwrite {
+		if _, err = os.Stat(dstFileName); err == nil {
+			err = fmt.Errorf("destination file %s already present", dstFileName)
+			return
+		}
 	}
 
 	source, err := os.OpenFile(srcFileName, syscall.O_DIRECT, FilePermission_0600)
 	if err != nil {
-		err = fmt.Errorf("File %s opening error: %v", srcFileName, err)
+		err = fmt.Errorf("file %s opening error: %v", srcFileName, err)
 		return
 	}
 
 	// Closing file at the end.
 	defer CloseFile(source)
 
-	destination, err := os.OpenFile(newFileName, os.O_WRONLY|os.O_CREATE|syscall.O_DIRECT, FilePermission_0600)
+	var destination *os.File
+	if allowOverwrite {
+		destination, err = os.OpenFile(dstFileName, os.O_WRONLY|os.O_CREATE|syscall.O_DIRECT|os.O_TRUNC, FilePermission_0600)
+	} else {
+		destination, err = os.OpenFile(dstFileName, os.O_WRONLY|os.O_CREATE|syscall.O_DIRECT, FilePermission_0600)
+	}
+
 	if err != nil {
-		err = fmt.Errorf("Copied file creation error: %v", err)
+		err = fmt.Errorf("copied file creation error: %v", err)
 		return
 	}
+	if destination == nil {
```

Comment:
is this case possible where both err and destination are nil ?

Reply:
It's a pointer, so theoretically yes. Implementation-wise (of `os.OpenFile`) may be no. I've removed this check now. Will see later if we get an exception or some other issue.